### PR TITLE
CO-2750 Hr attendance management migration v11

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

**Comment by ecino** [comment]
hr_attendance_management/models/hr_employee.py:86:89: E501 line too long (92 > 88 characters)
@eicher31 please review

File: hr_attendance_management/data/annual_balance_cron.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<odoo>
+    <data noupdate="1">
+        <record id="ir_cron_compute_annual_balance"
+                model="ir.cron">
+            <field name="name">Compute annual extra hours balance</field>
+            <field name="interval_type">months</field>
+            <field name="interval_number">12</field>
+            <field name="numbercall">-1</field>
+            <field name="model_id" ref="model_hr_employee"/>
+            <field name="function">_cron_compute_annual_balance</field>
```
Comment (eicher31) [inline hr_attendance_management/data/annual_balance_cron.xml]: Verify if it work:
2020-01-24 10:33:17,166 3357 WARNING v11c_hr_attendance odoo.models: ir.cron.create() includes unknown fields: function
Seem to be replace by the field "code":
https://webkul.com/blog/creating-cron-server-action-odoo-11/

File: hr_attendance_management/data/attendance_computation_cron.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <data noupdate="1">
+        <record id="ir_cron_create_att_comp_result" model="ir.cron">
+            <field name="name">Create attendance computation results</field>
+            <field name="interval_type">days</field>
+            <field name="numbercall">-1</field>
+            <field name="model_id" ref="model_hr_employee"/>
+            <field name="function">_cron_create_attendance</field>
```
Comment (eicher31) [inline hr_attendance_management/data/attendance_computation_cron.xml]: Same

File: hr_attendance_management/views/hr_employee.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    <record model="ir.ui.view" id="hr_employee_extra_hours_kanban">
+        <field name="name">HR Employee Extra Hours Kanban View</field>
+        <field name="model">hr.employee</field>
+        <field name="priority"
+               eval="20"/>
+        <field name="inherit_id"
+               ref="hr.hr_kanban_view_employees"/>
+        <field name="arch"
+               type="xml">
+            <xpath expr="//div[@class='oe_kanban_details']/ul/li" position="after">
```
Comment (eicher31) [inline hr_attendance_management/views/hr_employee.xml]: Use the hasclass(*classes) function to filter elements by their classes

**Review by eicher31 — CHANGES_REQUESTED** [review]
FAIL: test_period_balances (odoo.addons.hr_attendance_management.tests.test_periods.TestPeriod)                                                   
Traceback (most recent call last):                                                                                                                
`   File "/home/stephane/odoo/odoo11/review/compassion-modules/hr_attendance_management/tests/test_periods.py", line 224, in test_period_balances 
`     self.assertEquals(self.gilles.balance, 4)                                                                                                   
` AssertionError: 5.0 != 4

**Comment by ecino** [comment]
I think you should rebase onto branch 11.0 in order to fix the tests.

File: hr_attendance_management/tests/test_periods.py [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+
+        cls.period1 = cls.env['hr.employee.period'].create({
+            'start_date': cls.start_date_1,
+            'end_date': cls.end_date_1,
+            'balance': 1,
+            'previous_period': None,
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
Comment (eicher31) [inline hr_attendance_management/tests/test_periods.py]: WARNING v11c_hr_attendance_test odoo.models: hr.employee.period.create() includes unknown fields: origin

File: hr_attendance_management/tests/test_periods.py [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+
+        cls.period2 = cls.env['hr.employee.period'].create({
+            'start_date': cls.start_date_2,
+            'end_date': cls.end_date_2,
+            'balance': 2,
+            'previous_period': cls.period1.id,
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
Comment (eicher31) [inline hr_attendance_management/tests/test_periods.py]: WARNING v11c_hr_attendance_test odoo.models: hr.employee.period.create() includes unknown fields: origin

File: hr_attendance_management/tests/test_periods.py [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+
+        cls.period3 = cls.env['hr.employee.period'].create({
+            'start_date': cls.start_date_3,
+            'end_date': cls.end_date_3,
+            'balance': 3,
+            'previous_period': cls.period2.id,
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
Comment (eicher31) [inline hr_attendance_management/tests/test_periods.py]: same

**Review by eicher31 — CHANGES_REQUESTED** [review]
Move the hr_attendance settings inside the general settings and update the view. Some fields have not label.

**Comment by quentingigon** [comment]
There is a problem with advanced_translation: it seems it can't launch the tests and after that the execution stops. Is it a problem with Travis ?

**Comment by quentingigon** [comment]
I removed a dependency on partner-contact-gender as it was causing Travis to fail. This module did add the gender field to res.partner but the values were (male, Male) instead of (m, Male). I replaced the dependency by thankyou_letter, as it's this module that add the gender field

**Review by eicher31 — APPROVED** [review]

**Pull request merged**